src/coreclr/gc/gcpriv.h

```diff
@@ -5508,6 +5508,8 @@ class heap_segment
     // We may keep per region free list later which requires more work.
     uint8_t*        free_list_head;
     uint8_t*        free_list_tail;
+    size_t          free_list_size;
+    size_t          free_obj_size;
```

Comment:
could add a comment here and mention that these are only used by SIP regions

Reply:
You already had a comment above the declaration of free_list_head that explains this and would naturally apply to the two new fields as well.